include/deal.II/matrix_free/dof_info.templates.h

```diff
@@ -437,11 +437,14 @@ namespace internal
                       unsigned int *data_ptr =
                         plain_dof_indices.data() +
                         row_starts_plain_indices[boundary_cells[i]];
+                      const unsigned int index =
+                        (cell_active_fe_index.size() == 0 ||
+                         dofs_per_cell.size() == 1) ?
+                          0 :
+                          cell_active_fe_index[i];
```

Comment:
This looks indeed right. In the other functions there is a check `have_hp` of this kind, I guess you took it from there. It might be that you hit more bugs of this kind as I have not tested the categorization on mg levels, but this seems to be the way to go.

Reply:
yes, i picked it up from other places in DoF info, testing this code now...